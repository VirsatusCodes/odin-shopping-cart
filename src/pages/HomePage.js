import React, {useState} from 'react';

function HomePage({productList}) {
  return (
    <div>
        {<img src={productList.item1.img} alt = 'deoderant'/>}
    </div>
  );
}

export default HomePage;
