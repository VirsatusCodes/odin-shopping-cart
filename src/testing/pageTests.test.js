import React from "react";
import {render, screen, act, waitForElementToBeRemoved} from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import HomePage from "../pages/HomePage";


describe('HomePage testing', ()=> {

    it('renders greeting', () => {
        render(<HomePage />); /* one instance of render HomePage cant work for multiple tests apparently */
        
        expect(screen.getByRole('heading').textContent).toMatch(/Welcome to our store!/);
    });

    it('renders mission statement', () => {
        render(<HomePage />); 
        
        expect(screen.getByText(/Here/)).toBeInTheDocument();
    })
});