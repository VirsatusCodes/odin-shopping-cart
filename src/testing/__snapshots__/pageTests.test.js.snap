// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalogue testing renders a div w/correct info 1`] = `
<DocumentFragment>
  <div>
    <ul
      class="catalogue-of-items"
    />
  </div>
</DocumentFragment>
`;
